DataStructures/LinkedList.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace LinkedList
+{
+
+    public class LinkedList<T> : IEnumerable<T>
+    {
+        private Node<T> _head;
+
+        public int Length { get; private set; }
+
+        private Node<T> _lastAdd { get; set; }
+
+
+
+
+        public LinkedList() { }
+
+        public void Add(T value)
+        {
+            Node<T> current;
+            if (_head == null)
+            {
+                _head = new Node<T>(value);
+                Length++;
+                _lastAdd = _head;
+                return;
+            }
+
+            current = _head;
+            while (current.Next != null)
+            {
+                current = current.Next;
+            }
+
+            current.Next = new Node<T>(value);
+            Length++;
+            _lastAdd = current.Next;
+
+        }
+
+        public void Pop()
+        {
+            var current = _head;
+            if (_lastAdd == _head)
+            {
+                var currentNode = _head.Next;
+                _head = currentNode;
+                _lastAdd = _head;
+                Length--;
+                return;
+            }
+
+            while (current.Next != _lastAdd)
```

Comment:
Looks like searching for a new head. Please add comments ar extract separate method